### ISSUE TITLE
[Documentation][2.0] Need more info about adding new compiler

### ISSUE THREAD

**Issue by Kakadu** [ocaml/opam#2716]
It seems that now it is not as obvious as I expected. I tried to do this manually using fresh local repository but my new compiler switch is not being detected. The only thing I read is https://opam.ocaml.org/blog/opam-2-0-preview/ but it is not enough for me.

Maybe we can have these guidelines on separated man page?

```
➜  opam-repo-local find .                                                          
.
./urls.txt
./packages
./packages/ocaml-variants
./packages/ocaml-variants/ocaml-variants.4.03.0+effects
./packages/ocaml-variants/ocaml-variants.4.03.0+effects/opam
./index.tar.gz
➜  opam-repo-local cat ./packages/ocaml-variants/ocaml-variants.4.03.0+effects/opam
opam-version: "2.0"
synopsis: "ocaml 4.03 with effects"
maintainer: "kakadu.hafanana"
depends: ["base-unix" "base-bigarray" "base-threads"]
flags: compiler
setenv: CAML_LD_LIBRARY_PATH = "%{lib}%/stublibs"
build: [
  ["mkdir" "-p" "%{lib}%/ocaml/"]
  [
    "sh"
    "-exc"
    "echo \"* : g = 1\" > %{lib}%/ocaml/ocaml_compiler_internal_params"
  ]
  ["./configure" "-prefix" prefix]
  [make "world.opt"]
]
install: [make "install"]
url {
  src: "https://github.com/ocamllabs/ocaml-effects/archive/master.zip"
}
➜  opam-repo-local opam remote

=-=- Repository configuration for switch 4.03.0 -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
 1 local   file:///home/kakadu/opam-repo-local
 2 default https://opam.ocaml.org/2.0~dev

➜  opam-repo-local opam update local

=-=- Updating package repositories =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
[local] file:///home/kakadu/opam-repo-local synchronized
Now run 'opam upgrade' to apply any package updates.
➜  opam-repo-local opam switch list-available --all |grep effects         
➜  opam-repo-local
```

**Comment by AltGr** [comment]
I confirm that you seem to have gotten it right.
The trouble you got into, it seems, may be due to per-switch repositories: if you look at the output of `opam remote`, it says "Repository configuration __for switch 4.04.0__". Try `opam repo -a` to get the list of configured repositories for each switch, and check what is set or not as default.

Note that the `opam repo` interface was simplified a lot for handling per-switch repos very recently (i.e. this week). We should manage to avoid this trap in the UI though: we could
- remove the "default repo setting" altogether, it is the part that is bothering me; we could simply copy the current switch setting (avoids the confusion, but it doesn't feel good that `switch create` may depend on the switch you were in. And what if you weren't in one ?) ; or force the user to specify the repositories, if more than one was configured.
- at least, we could advertise the repositories that are being used on `switch create` and `list-available`, that would alleviate the confusion. Would that have been enough in your case to avoid the problem ?

In short, you could do:
- `opam repo add --set-default local`, then `list-available`, or
- `opam switch list-available --all --repos local,default`

**Comment by AltGr** [comment]
Forgot to link to the 2.0 FAQ:

  https://github.com/ocaml/opam/wiki/opam-2.0-FAQ
